Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Multistage docker build, requires docker 17.05
+
+# builder stage
+FROM nvidia/cuda:10.0-devel as builder
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get --no-install-recommends --yes install \
+        libncurses5-dev \
+        libncursesw5-dev \
+        cmake \
+        git \
+        curl
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+RUN git clone https://github.com/mimblewimble/grin-miner && cd grin-miner && git submodule update --init
```

Comment:
I suggest to use `git clone --recursive` to avoid the `git submodule update --init`

Reply:
I did it that way because it is how the documentation says and it will be easy for someone to follow it if is going over it at the same time than over the build documentation.